apps/meteor/server/services/moderation/service.ts

```diff
@@ -0,0 +1,74 @@
+import { ServiceClassInternal } from '@rocket.chat/core-services';
+import type { IModerationService } from '@rocket.chat/core-services';
+import { Permissions, Roles, Users } from '@rocket.chat/models';
+
+import { createOrUpdateProtectedRoleAsync } from '../../lib/roles/createOrUpdateProtectedRole';
+import { explorerPermissions as explorerExplicitPermissions, novicePermissions, trustRoles } from '../../../app/moderation/lib/permissions';
+import { Logger } from '../../lib/logger/Logger';
+
+export class ModerationService extends ServiceClassInternal implements IModerationService {
+	protected name = 'moderation';
+
+	private logger = new Logger('Moderation');
+
+	async removeTrustRoles(): Promise<void> {
+		const role = await Roles.findOneById(trustRoles[0], { projection: { _id: 1 } });
+		if (!role) {
+			return;
+		}
+		// remove all role mentions from user records that have it
+		// TODO: move to models
+		try {
+			await Users.updateMany({ roles: { $in: trustRoles } }, { $pullAll: { roles: trustRoles as unknown as string[] } });
+			await Permissions.updateMany({ roles: { $in: trustRoles } }, { $pullAll: { roles: trustRoles as unknown as string[] } });
```

Comment:
These queries should be on model :)

Reply:
Thank you so much for the review. I agree, will move it!